Pinta.Gui.Widgets/Widgets/Canvas/CanvasRenderer.cs

```diff
@@ -103,13 +103,28 @@ public void Render (
 			g.Restore ();
 		}
 
-		// If we are at least 200% and grid is requested, draw it
-		if (enable_pixel_grid && PintaCore.Actions.View.PixelGrid.Value && scale_factor.Ratio <= 0.5d)
+		// Draw the grid if it's enabled and we're zoomed in far enough
+		const int minGridLineDistance = 5;
+		if (enable_grid && PintaCore.CanvasGrid.ShowGrid && GetMinGridLineDistance () >= minGridLineDistance)
```

Comment:
Could the canvas renderer receive an `ICanvasGridService` to eliminate the global PintaCore dependency?
This could actually also eliminate the `enable_grid` member too if you use a nullable `ICanvasGridService?` to indicate that the canvas grid should never be displayed (e.g. for the preview images in the layers list view widget)